Core/Object Arts/Dolphin/Base/BlockClosure.cls

```diff
@@ -549,6 +549,33 @@ stackTempCount: anInteger
 	info := (info bitAnd: ##((16rFF << 15) bitInvert)) 
 				bitOr: ((anInteger bitAnd: 16rFF) bitShift: 15)!
 
+storeOn: aStream
+	"Append, to aStream, a <String> whose characters can recreate the receiver."
+
+	| m block class selector string |
+	m := self method.
+	(m notNil and: [m isExpression]) ifTrue: [	"We can print the source"
+		aStream 
+			nextPut: $(;
+			nextPutAll: m getSource;
+			nextPut: $).
+		^self].
+	m ifNil: [^super storeOn: aStream].
+	class := m methodClass.
+	selector := m selector.
+	(class isKindOf: Class) ifTrue: [
```

Comment:
Instantiating objects and evaluating arbitrary methods could have unknown and undesirable side effects. This doesn't seem appropriate for a general purpose #storeOn: implementation, so I don't think you can do this. Also there are many cases where this could fail (object cannot be instantiated with new, not a zero arg selector).